app/services/username_generator.rb

```diff
@@ -2,14 +2,19 @@ module UsernameGenerator
 
   def self.run(first_name, last_name, classcode)
     part1 = "#{first_name}.#{last_name}"
-    part1_pattern = "%#{part1}%"
     at_classcode = self.at_classcode(classcode)
-    extant = User.where("username ILIKE ?", part1_pattern)
+    pattern = "#{part1}[0-9]+#{at_classcode}"
+    extant = User.where("username SIMILAR TO ?", pattern)
```

Comment:
I believe @marcellosachs knew of a bug that occurred when similarly anmed students are added to a class. perhaps this is a fix for it.